crates/polars-plan/src/plans/aexpr/schema.rs

```diff
@@ -116,49 +123,59 @@ impl AExpr {
                             out_field.name()
                         };
                         Field::new(out_name.clone(), Boolean)
                     },
                     Operator::TrueDivide => {
-                        return get_truediv_field(*left, *right, arena, ctx, schema, nested)
+                        return get_truediv_field(*left, *right, arena, ctx, schema, agg_list)
                     },
                     _ => {
-                        return get_arithmetic_field(*left, *right, arena, *op, ctx, schema, nested)
+                        return get_arithmetic_field(
+                            *left, *right, arena, *op, ctx, schema, agg_list,
+                        )
                     },
                 };
 
                 Ok(field)
             },
-            Sort { expr, .. } => arena.get(*expr).to_field_impl(schema, ctx, arena, nested),
+            Sort { expr, .. } => arena.get(*expr).to_field_impl(schema, ctx, arena, agg_list),
             Gather {
                 expr,
                 returns_scalar,
                 ..
             } => {
                 if *returns_scalar {
-                    *nested = nested.saturating_sub(1);
+                    *agg_list = false;
                 }
-                arena.get(*expr).to_field_impl(schema, ctx, arena, nested)
+                arena
+                    .get(*expr)
+                    .to_field_impl(schema, ctx, arena, &mut false)
             },
-            SortBy { expr, .. } => arena.get(*expr).to_field_impl(schema, ctx, arena, nested),
-            Filter { input, .. } => arena.get(*input).to_field_impl(schema, ctx, arena, nested),
+            SortBy { expr, .. } => arena.get(*expr).to_field_impl(schema, ctx, arena, agg_list),
+            Filter { input, .. } => arena
+                .get(*input)
+                .to_field_impl(schema, ctx, arena, agg_list),
             Agg(agg) => {
                 use IRAggExpr::*;
                 match agg {
                     Max { input: expr, .. }
                     | Min { input: expr, .. }
                     | First(expr)
                     | Last(expr) => {
-                        *nested = nested.saturating_sub(1);
-                        arena.get(*expr).to_field_impl(schema, ctx, arena, nested)
+                        *agg_list = false;
+                        arena
+                            .get(*expr)
+                            .to_field_impl(schema, ctx, arena, &mut false)
                     },
                     Sum(expr) => {
-                        *nested = nested.saturating_sub(1);
-                        let mut field =
-                            arena.get(*expr).to_field_impl(schema, ctx, arena, nested)?;
+                        *agg_list = false;
+                        let mut field = arena
+                            .get(*expr)
+                            .to_field_impl(schema, ctx, arena, &mut false)?;
```

Comment:
We don't pass `agg_list` upwards anymore as a `sum()` would always return a scalar.